### ISSUE TITLE
"Add to campaign" on versioned GridFields

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-campaign-admin#89]
# Overview

At the moment you can only add pages and files to campaigns via the UI. Other DataObjects can be implicitly added through a custom `$owns` relationship on a page or a file, but not separately through the UI. 

# Acceptance Criteria

 * I can add a DataObject to an existing or new campaign when viewing it in ModelAdmin
 * I can add a DataObject to an existing or new campaign when viewing it in GridField (e.g. nested in the pages section)
 * I can find out about these capabilities through https://docs.silverstripe.org/en/4/developer_guides/model/versioning/

# Related

 * [Make it easier to add items to a campaign](https://github.com/silverstripe/silverstripe-campaign-admin/issues/40)

**Comment by chillu** [comment]
@bergice Yep, any dataobject. It might not have a way to be previewed within the campaign admin (unless defined by the dev?), but that should be handled gracefully already in the UI.

**Comment by clarkepaul** [comment]
@chillu FYI I've added this to backlog/discussion pipeline as this should be something we do (along with the ability to add content blocks to a campaign without the page it belongs too).

**Comment by KINKCreative** [comment]
Has there been any movement on this? Are there other workarounds to add a custom DataObject to a Campaign?

**Comment by UndefinedOffset** [comment]
Not sure if there's been movement on this side but I put something together on webbuilders-group/silverstripe-add-to-campaigns we've used it on a few projects and it seems to work well.
